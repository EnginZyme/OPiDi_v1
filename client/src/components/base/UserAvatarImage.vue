<template>
  <img
    :src="userImageURL"
    alt="Avatar"
    :class="['user-avatar-image', `user-avatar-image--${size}`]"
  />
</template>

<script>
import { mapGetters } from "vuex";

/**
 * A component to display the image of the authenticated user.
 */
export default {
  name: "UserAvatarImage",
  props: {
    /**
     * Determines the size of the rendered image.
     * 
     * @values small, large
     */
    size: {
      type: String,
      default: "small",
    },
  },
  computed: {
    ...mapGetters(["userImageURL"]),
  },
};
</script>

<style scoped>
.user-avatar-image {
  align-self: center;
  vertical-align: middle;
  width: 32px;
  height: 32px;
  border-radius: 50%;
}
.user-avatar-image--large {
  width: 80px;
  height: 80px;
}
</style>
