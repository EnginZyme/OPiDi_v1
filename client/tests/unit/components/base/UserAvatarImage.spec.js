import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import UserAvatarImage from "@/components/base/UserAvatarImage";

const localVue = createLocalVue();

localVue.use(Vuex);

const sampleImageSrc = "../../assets/logo_dark.svg";

describe("shallowMounted UserAvatarImage", () => {
  let getters;
  let store;
  let wrapper;

  beforeEach(() => {
    getters = {
      userImageURL: () => sampleImageSrc,
    };

    store = new Vuex.Store({
      getters,
    });

    wrapper = shallowMount(UserAvatarImage, {
      store,
      localVue,
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe("shallowMounted UserAvatarImage with 'size' prop set to large", () => {
  let getters;
  let store;
  let wrapper;

  beforeEach(() => {
    getters = {
      userImageURL: () => sampleImageSrc,
    };

    store = new Vuex.Store({
      getters,
    });

    wrapper = shallowMount(UserAvatarImage, {
      store,
      localVue,
      propsData: {
        large: true,
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
