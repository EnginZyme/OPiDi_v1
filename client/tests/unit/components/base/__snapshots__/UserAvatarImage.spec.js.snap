// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted UserAvatarImage renders correctly 1`] = `
<img
  alt="Avatar"
  class="user-avatar-image user-avatar-image--small"
  src="../../assets/logo_dark.svg"
/>
`;

exports[`shallowMounted UserAvatarImage with 'size' prop set to large renders correctly 1`] = `
<img
  alt="Avatar"
  class="user-avatar-image user-avatar-image--small"
  large="true"
  src="../../assets/logo_dark.svg"
/>
`;
